### PR TITLE
Remove plugins xml

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Port PR: https://github.com/openvinotoolkit/openvino/pull/15489

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: cmake/developer_package/plugins/plugins.hpp.in [hunk 1]
```diff
@@ -21,3 +26,16 @@ inline const std::map<Key, Value> getStaticPluginsRegistry() {
...
 }
+
+#else
+
+using Key = std::string;
+using Value = std::string;
+using PluginsCompiledRegistry = std::map<Key, Value>;
+
+inline const std::map<Key, Value> getCompiledPluginsRegistry() {
+@IE_PLUGINS_COMPILED@
```
Comment (ilya-lavrenov) [inline cmake/developer_package/plugins/plugins.hpp.in:37]: why do we still need to have a conditional compilation? 
what if I used `ENABLE_PLUGINS_XML` here?

File: src/inference/src/dev/core_impl.hpp [hunk 2]
```diff
@@ -233,11 +227,35 @@ class CoreImpl : public InferenceEngine::ICore, public std::enable_shared_from_t
...
+        for (const auto& plugin : plugins) {
+            const auto& deviceName = plugin.first;
+            const auto& pluginPath = ov::util::get_compiled_plugin_path(plugin.second);
+
+            if (pluginRegistry.find(deviceName) == pluginRegistry.end() && FileUtils::fileExist(pluginPath)) {
+                PluginDescriptor desc{pluginPath};
+                pluginRegistry[deviceName] = desc;
+                add_mutex(deviceName);
+            }
         }
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/core_impl.hpp:259]: can we move definitions of these methods to cpp?

File: cmake/developer_package/plugins/create_plugins_hpp.cmake [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-foreach(var IE_DEVICE_MAPPING IE_PLUGINS_HPP_HEADER IE_PLUGINS_HPP_HEADER_IN)
+foreach(var IE_DEVICE_MAPPING OV_DYNAMIC IE_PLUGINS_HPP_HEADER IE_PLUGINS_HPP_HEADER_IN)
```
Comment (ilyachur) [inline cmake/developer_package/plugins/create_plugins_hpp.cmake:5]: In future we can rename ie_plugins -> ov_plugins and change prefix `IE` to `OV` in functions

File: src/inference/src/dev/core_impl.hpp [hunk 4]
```diff
@@ -15,17 +15,15 @@
...
 #include "openvino/core/extension.hpp"
 #include "openvino/core/version.hpp"
 #include "openvino/runtime/common.hpp"
 #include "openvino/runtime/icompiled_model.hpp"
 #include "openvino/runtime/threading/executor_manager.hpp"
-
-#ifdef OPENVINO_STATIC_LIBRARY
-#    include "ie_plugins.hpp"
-#endif
+#include "openvino/util/file_util.hpp"
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/core_impl.hpp:26]: looks like we don't need this and `ie_plugins.hpp` include header

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilyachur** [comment]
I will fix all comments in the next PR.

**Pull request merged**